### PR TITLE
feat: React circle CI setup

### PR DESCRIPTION
**Pull request by shefalig972**
Actually I followed the steps what you have mentioned in your comment in earlier PR.
But there were some commits coming after HARD RESET.

So keep it simple, I just remove the earlier CI branch and PR.
Took the fresh clone of Develop Branch and Created a new feature branch of circle CI from develop.
Then committed the Circle CI code in that  circle CI feature branch and pushed the code.
So please review and merge.

### CODE REVIEW

**Comment by programmingfullstack** [comment]
Circle CI Integration is successfully implemented, So just commented the 'yarn start' command as it is making CI "Running".

So to make the Circle CI job a "Success", just commented 'yarn start' command.
You can also see the screen-shot below:-

https://user-images.githubusercontent.com/50453389/57533197-d1770580-735a-11e9-91b9-abd8d21f73ce.png

Please review and merge the PR

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            command: sudo curl -sLf -o /dev/null 'https://deb.nodesource.com/node_8.x/dists/jessie/Release'    
+        - run : 
+            name: Making curl request to update node source list
+            command: sudo curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | sudo apt-key add -
+        - run: 
+            name: Get into source list
+            command: sudo cp /etc/apt/sources.list ~/ 
+        - run: 
+            name: get Source list
+            command: sudo wget "http://pastebin.com/raw.php?i=uzhrtg5M" -O /etc/apt/sources.list
```
Comment (br0wn) [inline .circleci/config.yml]: Executing a hosted unsecured script with sudo is an extreme security issue! Please remove this.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+jobs: # a collection of steps
+  build: # runs not using Workflows must have a `build` job as entry point
+    working_directory: ~/tazman-frontend # directory where steps will run
+    docker: # run the steps with Docker
+        - image: circleci/node:latest # ...with this image as the primary container; this is where all `steps` will run
+    steps: # a collection of executable commands
+        - checkout # special step to check out source code to working directory        
+        - run:
+            name: Initializing the yarn installation in container
+            command: 'curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add - && echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list'
```
Comment (br0wn) [inline .circleci/config.yml]: From the Circle CI [documentation](https://circleci.com/docs/2.0/yarn/):
```
Yarn might already be installed in your build environment if you are using the docker executor. With Pre-built CircleCI Docker Images, the NodeJS image (circleci/node) already has Yarn preinstalled.
```
so adding yarn sources is obsolete. Please remove this. 

Or was this necessary for some reason?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            command: sudo cp /etc/apt/sources.list ~/ 
+        - run: 
+            name: get Source list
+            command: sudo wget "http://pastebin.com/raw.php?i=uzhrtg5M" -O /etc/apt/sources.list 
+        - run: 
+            name: Get New updates
+            command: sudo apt-get update      
+        - run:
+            name: Installing yarn
+            command: sudo apt-get install yarn
```
Comment (br0wn) [inline .circleci/config.yml]: All of the steps from above should not be necessary as node is at latest version and yarn is already installed. Was there a specific reason why the sources are updated like this?

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        # - run:
+        #     name: test
+        #     command: yarn test
+        - restore_cache: 
+            key: dependency-cache-{{ checksum "package.json" }}
+        - run:
+            name: yarn install dependencies
+            command: yarn install
```
Comment (br0wn) [inline .circleci/config.yml]: Make this command like this
```
yarn install --frozen-lockfile
```

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            name: Get New updates
+            command: sudo apt-get update      
+        - run:
+            name: Installing yarn
+            command: sudo apt-get install yarn
+        # - run:
+        #     name: test
+        #     command: yarn test
+        - restore_cache: 
+            key: dependency-cache-{{ checksum "package.json" }}
```
Comment (br0wn) [inline .circleci/config.yml]: Instead of `package.json`, cache `yarn.lock`.

File: .circleci/config.yml (hunk 1 above)
Comment (programmingfullstack) [reply .circleci/config.yml]: Done.

File: .circleci/config.yml (hunk 2 above)
Comment (programmingfullstack) [reply .circleci/config.yml]: As earlier node was not latest version, it was not getting complied at that time...so As we have node latest version now I removed all the unwanted commands.

File: .circleci/config.yml (hunk 3 above)
Comment (programmingfullstack) [reply .circleci/config.yml]: As earlier node was not latest version, it was not getting complied at that time...so As we have node latest version now I removed all the unwanted commands.

File: .circleci/config.yml (hunk 4 above)
Comment (programmingfullstack) [reply .circleci/config.yml]: Done.

File: .circleci/config.yml (hunk 5 above)
Comment (programmingfullstack) [reply .circleci/config.yml]: Done.

**Comment by programmingfullstack** [comment]
I have done the changes as per your suggestion please review and merge the PR, thanks a lot!!

**Pull request merged**